src/frameworks/utils.ts

```diff
@@ -345,7 +345,19 @@ export function relativeRequire(dir: string, mod: string) {
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore prevent VSCE webpack from erroring on non_webpack_require
     const path = requireFunc.resolve(mod, { paths: [dir] });
-    if (extname(path) === ".mjs") {
+
+    let packageJson: PackageJson | undefined;
+    const isEsm =
+      extname(path) === ".mjs" ||
+      (packageJson = readJsonSync(join(dirname(path), "package.json"), { throws: false }))?.type ===
```

Comment:
@jamesdaniels would love to use the async readJson here but would need to handle the promise in all calls to relativeRequire, WDYT?

Reply:
i'm ok with us converting this to be async